cabal-install/src/Distribution/Client/IndexUtils.hs

```diff
@@ -1167,23 +1167,27 @@ readIndexCache verbosity index = do
 -- 'dieWithException's if it fails again). Throws IOException if any arise.
 readNoIndexCache :: Verbosity -> Index -> IO NoIndexCache
 readNoIndexCache verbosity index = do
-  cacheOrFail <- readNoIndexCache' index
+  cacheOrFail <- readNoIndexCache' verbosity index
   case cacheOrFail of
     Left msg -> do
       warn verbosity $
         concat
-          [ "Parsing the index cache failed ("
+          [ "Parsing the index cache for repo \""
+          , unRepoName (repoName repo)
+          , "\" failed ("
           , msg
           , "). "
           , "Trying to regenerate the index cache..."
           ]
 
       updatePackageIndexCacheFile verbosity index
 
-      either (dieWithException verbosity . CorruptedIndexCache) return =<< readNoIndexCache' index
+      either (dieWithException verbosity . CorruptedIndexCache) return =<< readNoIndexCache' verbosity index
 
     -- we don't hash cons local repository cache, they are hopefully small
     Right res -> return res
+  where
+    RepoIndex _ctxt repo = index
```

Comment:
I bet there is no way to use an accessor, right?

Reply:
No, there is no accessor at the moment.